src/domain/usecase/link-social-account.ts

```diff
@@ -0,0 +1,29 @@
+import * as uuid from 'node-uuid'
+
+import { UserRepository } from '../../application/repository'
+import { User } from '../model'
+
+interface Dependencies {
+  userRepository: UserRepository
+}
+
+export type LinkSocialAccount = (options: {
```

Comment:
If you guys can suggest a better name for this use case that would be great. Its adding users if they do not exist in the DB after social logins

Reply:
That's a fine name to me.